main.rb

```diff
@@ -0,0 +1,160 @@
+# enumerable methods
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    my_arr = *self
+    i = 0
+    while i < my_arr.length
+      yield my_arr[i]
+      i += 1
+    end
+    my_arr
+  end

```

Comment:
- [ ] my_each when a block is given when self is a range after calling the given block once for each element in self, returns the range itself. Your method is returning an array, even when calling my_each from a range

Example:
`(1..3).my_each { | value | puts value }` should print 1, 2 and 3, and after that should return the range itself `1..3`

Print from the original method:

![Screenshot from 2020-06-05 09-19-36](https://user-images.githubusercontent.com/30977845/83875559-b1f10e80-a70d-11ea-9870-7af5abdfb004.png)